src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -278,8 +278,13 @@ class Crystal::Doc::Generator
   end
 
   def compute_repository
+    `git rev-parse --is-inside-work-tree >/dev/null 2>&1`
+    unless $?.success?
+      puts "WARNING there is not inside git worktree. some features are not available."
+      return
+    end
+
     remotes = `git remote -v`
```

Comment:
`git remote` returns list of remotes with remote url after name (because of verbose `-v` switch). Adding `2>/dev/null` will silence error while leaving status code intact, thus triggering return on the next line.